golem/event_bus/base.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from typing import Awaitable, Callable, Generic, Optional, Type, TypeVar
+
+from golem.exceptions import GolemException
+
+TCallbackHandler = TypeVar("TCallbackHandler")
+
+TEvent = TypeVar("TEvent", bound="Event")
+
+
+class Event(ABC):
+    """Base class for all events."""
+
+
+class EventBusError(GolemException):
+    pass
+
+
+class EventBus(ABC, Generic[TCallbackHandler]):
```

Comment:
@approxit again, mostly out of curiosity - why is the callback handler type defined as a generic here? do we envision different signatures for specific callback handlers? wouldn't using just the `_CallbackInfo` type suffice?